src/java/main/org/apache/zookeeper/server/quorum/QuorumCnxManager.java

```diff
@@ -438,9 +447,18 @@ synchronized private boolean connectOne(long sid, InetSocketAddress electionAddr
         Socket sock = null;
         try {
              LOG.debug("Opening channel to server " + sid);
-             sock = new Socket();
-             setSockOpts(sock);
-             sock.connect(electionAddr, cnxTO);
+             if (self.isSslQuorum()) {
+                 SSLSocket sslSock = x509Util.createSSLSocket();
+                 setSockOpts(sslSock);
+                 sslSock.connect(electionAddr, cnxTO);
+                 sslSock.startHandshake();
```

Comment:
This is a synchronous call and will block connectOne until the hand shake finishes. Current cnx manager does blocking IO when handling connections so while hand shake is going on other peers will not be able to connect to this server. This might not be a big problem with carefully tuned initLimit / syncLimit, but I thought we might want mention this somewhere in doc or wiki.

An alternative would be making hand shake async so cnx manager does not block, but I am currently leaning towards not do this and instead use ZOOKEEPER-900/901 to fix the root cause.